modules/configuring-pipeline-run-workspaces.adoc

```diff
@@ -0,0 +1,208 @@
+
+.Prerequisites
+* You have logged in to Open Data Hub.
+* You have the appropriate roles and permissions to edit DSPAs in your projects.
+* You have a pipeline server configured in your data science project.
+
+.Procedure
+. Log in to the {openshift-platform} web console.
+. Navigate to your data science project namespace.
+. Click *Search* -> *Resources* and select *DataSciencePipelinesApplication* from the resource list.
+. Select your DSPA instance and click *YAML* to edit the configuration.
+. Add or modify the workspace configuration under the `spec.apiServer.workspace` field:
++
+[source,yaml]
+----
+apiVersion: datasciencepipelinesapplications.opendatahub.io/v1
+kind: DataSciencePipelinesApplication
+metadata:
+  name: sample-dspa
+  namespace: my-namespace
+spec:
+  dspVersion: v2
+  apiServer:
+    deploy: true
+    workspace:
+      volumeClaimTemplateSpec:
+        accessModes:
+        - ReadWriteOnce
+        storageClassName: standard-csi        
+  # ... other DSPA configuration
+----
++
+Where:
++
+* `apiServer.workspace.volumeClaimTemplateSpec`: Configures the PVC template for workspace volumes
+** `accessModes`: Specifies the access modes for workspace PVCs (options: `ReadWriteOnce`, `ReadWriteMany`, `ReadOnlyMany`)
+** `storageClassName`: Specifies the storage class to use for workspace PVCs
+
+. Click *Save* to apply the changes.
+
+.Verification
+To confirm your configuration, run the following sample pipeline. It writes a file into the workspace by using `dsl.WORKSPACE_PATH_PLACEHOLDER` and then reads the same file in a subsequent task:
+
+[source,python]
+----
+from kfp import dsl
+
+@dsl.component
+def write_to_workspace(workspace_path: str) -> str:
+    """Write a file to the workspace."""
+    import os
+
+    file_path = os.path.join(workspace_path, "data", "test_file.txt")
+    os.makedirs(os.path.dirname(file_path), exist_ok=True)
+
+    with open(file_path, "w") as f:
+        f.write("Hello from workspace!")
+
+    print(f"Wrote file to: {file_path}")
+    return file_path
+
+@dsl.component
+def read_from_workspace(file_path: str) -> str:
+    """Read a file from the workspace using the provided file path."""
+    import os
+
+    if os.path.exists(file_path):
+        with open(file_path, "r") as f:
+            content = f.read()
+        print(f"Read content from: {file_path}")
+        print(f"Content: {content}")
+        assert content == "Hello from workspace!"
+        return content
+
+    print(f"File not found at: {file_path}")
+    return "File not found"
+
+@dsl.pipeline(
+    name="pipeline-with-workspace",
+    description="A pipeline that demonstrates workspace functionality",
+    pipeline_config=dsl.PipelineConfig(
+        workspace=dsl.WorkspaceConfig(
+            size='1Gi',
+            kubernetes=dsl.KubernetesWorkspaceConfig(
+                pvcSpecPatch={
```

Comment:
We should document this but it shouldn't be in the sample pipeline because this section is after the DSPA configuring the defaults.